train_llama.sh

```diff
@@ -0,0 +1,336 @@
+#!/bin/bash
+
+# set -x
+
+export GPU_MAX_HW_QUEUES=2
+export TORCH_NCCL_HIGH_PRIORITY=1
+
+
+# parsing input arguments
+for ARGUMENT in "$@"
+do
+   KEY=$(echo $ARGUMENT | cut -f1 -d=)
+
+   KEY_LENGTH=${#KEY}
+   VALUE="${ARGUMENT:$KEY_LENGTH+1}"
+   export "$KEY"="$VALUE"
+done
+
+# Change for multinode config
+export CUDA_DEVICE_MAX_CONNECTIONS=1
+
+TEE_OUTPUT="${TEE_OUTPUT:-1}"
+NO_TORCH_COMPILE="${NO_TORCH_COMPILE:-0}"
+USE_FLASH_ATTN="${USE_FLASH_ATTN:-1}"
+NO_TRAINING="${NO_TRAINING:-0}" # NO_TRAINING=1: for computing metrics only
+ENABLE_PROFILING="${ENABLE_PROFILING:-0}"
+ENABLE_ROPE="${ENABLE_ROPE:-1}"
+ENABLE_MOCK_DATA="${ENABLE_MOCK_DATA:-1}"
+DUMMY_RUN="${DUMMY_RUN:-0}"
+ADD_TASK="${ADD_TASK:-0}"
+LABEL="${LABEL:-"test"}"
+LOG_DIR="profile/${LABEL}"
+echo "NO_TRAINING=$NO_TRAINING"
+
+CWD=`pwd`
+GPUS_PER_NODE=`python -c "import torch; print(torch.cuda.device_count())"`
+
+
+# Change for multinode config
+MASTER_ADDR=localhost
+MASTER_PORT=23731
+NNODES=1
+NODE_RANK=0
+WORLD_SIZE=$(($GPUS_PER_NODE*$NNODES))
+
+MODEL_SIZE="${MODEL_SIZE:-70}"
+TP="${TP:-8}"
+PP="${PP:-1}"
+MBS="${MBS:-2}"
+BS="${BS:-8}"
+SEQ_LENGTH="${SEQ_LENGTH:-4096}"
+TOTAL_ITERS="${TOTAL_ITERS:-4}"
+SEQ_PARALLEL="${SEQ_PARALLEL:-1}" 
+CONTI_PARAMS="${CONTI_PARAMS:-0}"
+OPTIMIZER="${OPTIMIZER:-sgd}"
+TE_FP16="${TE_FP16:-1}"
+
+
+export CUDA_DEVICE_MAX_CONNECTIONS=1
+
+EXPERIMENT_DIR="experiment"
+mkdir -p $EXPERIMENT_DIR
+
+CHECKPOINT_PATH=$EXPERIMENT_DIR/ckpts
+rm -rf $CHECKPOINT_PATH
+mkdir -p $CHECKPOINT_PATH
+DATA_DIR=$EXPERIMENT_DIR/data
+mkdir -p $DATA_DIR
+
+TOKENIZER_MODEL=$EXPERIMENT_DIR/tokenizer.model
+
+# Download the tokenizer model
+if ! [ -f "$TOKENIZER_MODEL" ]; then
+wget -O $TOKENIZER_MODEL https://huggingface.co/NousResearch/Llama-2-7b-chat-hf/resolve/main/tokenizer.model
+fi
+
+# Prepare the dataset
```

Comment:
Seperate out the dataset preparation into a different script, so that we dont' need to download it everytime or make some changes to this script. We can use it as is without any modifications.

Reply:
It seems that once the dataset is downloaded, it will not need to download it again when we rerun the script.